### PR TITLE
feat(editors/SingleLineDiagram): allow updating X/Y coordinates in SLD for Busbar/ConductingEquipment/PowerTransformer

### PR DESCRIPTION
**Pull request by dlabordus**
Extend existing Wizards for Busbar/ConductingEquipment/PowerTransformer in SLD to make it possible to change the X and Y Coordinate for a component. This will be reflected in the SLD directly.

Also made some changes to drawing the SLD to minimize the overlap of bays/busbars/powertransformers. If 2 bays overlap each other some click events won't be trigger, because the elements are hidden by the other bay overlapping the element.

### CODE REVIEW

**Comment by danyill** [comment]
Just did a little test of this and noticed that making coordinates negative causes unrecoverable problems and no more SLD - might be worth looking into.

**Comment by dlabordus** [comment]
> Just did a little test of this and noticed that making coordinates negative causes unrecoverable problems and no more SLD - might be worth looking into.

Hi @danyill, I also tried entering negative values and it works for me. It doesn't look very nice, because items will move strange, but are displayed. But I will change the code to use 0 in these cases.

**Review by JakobVogelsang — APPROVED** [review]
In general LGTM. 

There is just one little thing regarding import statement

File: test/unit/editors/singlelinediagram/foundation.test.ts [hunk 1]
```diff
@@ -6,8 +6,9 @@ import {
   getPathNameAttribute,
   isBusBar,
   getConnectedTerminals,
-  calculateConnectivityNodeCoordinates,
+  calculateConnectivityNodeCoordinates, getCommonParentElement,
 } from '../../../../src/editors/singlelinediagram/foundation.js';
+import {getFixedCoordinateValue} from "../../../../src/editors/singlelinediagram/wizards/foundation";
```
[the remark this answers is not in the document]
Comment (dlabordus) [reply test/unit/editors/singlelinediagram/foundation.test.ts]: Fixed it. Also saw that the tests where in the wrong foundation.test.ts, so added a new one for these new tests.

**Pull request merged**